### PR TITLE
[TEST] use EXPECT_SAME_TYPE macro (oneline regexp)

### PR DESCRIPTION
**Pull request by marehr**
Search and replace with `EXPECT_TRUE\(\(std::same_as<(.+)>\)\);` and `EXPECT_SAME_TYPE($1);`

### CODE REVIEW

File: test/unit/alignment/configuration/align_config_score_type_test.cpp [hunk 1]
```diff
@@ -13,6 +13,8 @@
 #include <seqan3/core/configuration/configuration.hpp>
 #include <seqan3/utility/type_traits/basic.hpp>
 
+#include <seqan3/test/expect_same_type.hpp>
```
Comment (marehr) [inline test/unit/alignment/configuration/align_config_score_type_test.cpp:16]:
```suggestion
#include <seqan3/test/expect_same_type.hpp>
#include <seqan3/utility/type_traits/basic.hpp>
```

File: test/unit/alignment/matrix/alignment_coordinate_test.cpp [hunk 2]
```diff
@@ -12,28 +12,30 @@
 #include <range/v3/view/iota.hpp>
 
 #include <seqan3/alignment/matrix/alignment_coordinate.hpp>
+
```
Comment (marehr) [inline test/unit/alignment/matrix/alignment_coordinate_test.cpp:15]:
```suggestion
```

File: test/unit/alignment/matrix/debug_matrix_test.cpp [hunk 3]
```diff
@@ -11,6 +11,8 @@
 #include <seqan3/alignment/matrix/trace_directions.hpp>
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
```
Comment (marehr) [inline test/unit/alignment/matrix/debug_matrix_test.cpp:13]:
```suggestion
```

File: test/unit/alignment/matrix/detail/affine_cell_proxy_test.cpp [hunk 4]
```diff
@@ -11,6 +11,8 @@
 #include <seqan3/core/tuple/common_tuple.hpp>
 #include <seqan3/test/expect_same_type.hpp>
```
Comment (marehr) [inline test/unit/alignment/matrix/detail/affine_cell_proxy_test.cpp:13]:
```suggestion
```

File: test/unit/alignment/pairwise/align_pairwise_test.cpp [hunk 5]
```diff
@@ -21,6 +21,8 @@
 #include <seqan3/range/views/to.hpp>
 #include <seqan3/utility/tuple/concept.hpp>
```
Comment (marehr) [inline test/unit/alignment/pairwise/align_pairwise_test.cpp:23]:
```suggestion
```

File: test/unit/alignment/pairwise/align_pairwise_test.cpp [hunk 6]
```diff
@@ -21,6 +21,8 @@
...
 
+#include <seqan3/test/expect_same_type.hpp>
```
Comment (marehr) [inline test/unit/alignment/pairwise/align_pairwise_test.cpp:24]:
```suggestion
#include <seqan3/test/expect_same_type.hpp>
#include <seqan3/utility/tuple/concept.hpp>
```

File: test/unit/alignment/pairwise/alignment_configurator_test.cpp [hunk 7]
```diff
@@ -17,6 +17,8 @@
 #include <seqan3/range/views/zip.hpp>
 #include <seqan3/std/ranges>
```
Comment (marehr) [inline test/unit/alignment/pairwise/alignment_configurator_test.cpp:19]:
```suggestion
```

File: test/unit/alignment/pairwise/alignment_result_test.cpp [hunk 8]
```diff
@@ -22,6 +22,8 @@
 #include <seqan3/range/views/to_char.hpp>
 #include <seqan3/range/views/to.hpp>
```
Comment (marehr) [inline test/unit/alignment/pairwise/alignment_result_test.cpp:24]:
```suggestion
```

File: test/unit/alignment/pairwise/detail/type_traits_test.cpp [hunk 9]
```diff
@@ -12,6 +12,8 @@
 #include <seqan3/alignment/pairwise/detail/type_traits.hpp>
 #include <seqan3/std/concepts>
```
Comment (marehr) [inline test/unit/alignment/pairwise/detail/type_traits_test.cpp:14]:
```suggestion
```

File: test/unit/alignment/pairwise/pairwise_alignment_single_callback_test_template.hpp [hunk 10]
```diff
@@ -12,7 +12,9 @@
 #include <seqan3/alignment/configuration/align_config_score_type.hpp>
 #include <seqan3/alignment/pairwise/align_pairwise.hpp>
 #include <seqan3/range/views/to_char.hpp>
+
```
Comment (marehr) [inline test/unit/alignment/pairwise/pairwise_alignment_single_callback_test_template.hpp:15]:
```suggestion
```

File: test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp [hunk 11]
```diff
@@ -13,7 +13,9 @@
 #include <seqan3/alignment/pairwise/align_pairwise.hpp>
 #include <seqan3/range/views/to_char.hpp>
 #include <seqan3/range/views/to.hpp>
+
```
Comment (marehr) [inline test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp:16]:
```suggestion
```

File: test/unit/core/algorithm/pipeable_config_element_test_template.hpp [hunk 12]
```diff
@@ -14,6 +14,8 @@
 #include <seqan3/core/algorithm/pipeable_config_element.hpp>
 #include <seqan3/core/configuration/configuration.hpp>
```
Comment (marehr) [inline test/unit/core/algorithm/pipeable_config_element_test_template.hpp:16]:
```suggestion
```

File: test/unit/core/configuration/configuration_test.cpp [hunk 13]
```diff
@@ -10,7 +10,9 @@
 #include <range/v3/algorithm/equal.hpp>
 
 #include <seqan3/core/configuration/configuration.hpp>
+
```
Comment (marehr) [inline test/unit/core/configuration/configuration_test.cpp:13]:
```suggestion
```

File: test/unit/core/detail/iterator_traits_test.cpp [hunk 14]
```diff
@@ -16,6 +16,8 @@
 
 #include <seqan3/core/detail/iterator_traits.hpp>
```
Comment (marehr) [inline test/unit/core/detail/iterator_traits_test.cpp:18]:
```suggestion
```

File: test/unit/io/alignment_file/sam_tag_dictionary_test.cpp [hunk 15]
```diff
@@ -12,6 +12,8 @@
 #include <seqan3/io/alignment_file/sam_tag_dictionary.hpp>
 #include <seqan3/std/concepts>
```
Comment (marehr) [inline test/unit/io/alignment_file/sam_tag_dictionary_test.cpp:14]:
```suggestion
```

File: test/unit/range/container/bitcompressed_vector_test.cpp [hunk 16]
```diff
@@ -11,7 +11,9 @@
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
 #include <seqan3/range/container/bitcompressed_vector.hpp>
 #include <seqan3/range/views/complement.hpp>
+
```
Comment (marehr) [inline test/unit/range/container/bitcompressed_vector_test.cpp:14]:
```suggestion
```

File: test/unit/range/container/small_vector_test.cpp [hunk 17]
```diff
@@ -13,7 +13,9 @@
 #include <seqan3/range/container/concept.hpp>
 #include <seqan3/range/container/small_vector.hpp>
 #include <seqan3/std/ranges>
+
```
Comment (marehr) [inline test/unit/range/container/small_vector_test.cpp:16]:
```suggestion
```

File: test/unit/range/views/drop_test.cpp [hunk 18]
```diff
@@ -23,7 +23,9 @@
 #include <seqan3/std/algorithm>
 #include <seqan3/std/concepts>
 #include <seqan3/std/ranges>
+
```
Comment (marehr) [inline test/unit/range/views/drop_test.cpp:26]:
```suggestion
```

File: test/unit/range/views/repeat_test.cpp [hunk 19]
```diff
@@ -16,18 +16,20 @@
 #include <seqan3/range/views/repeat.hpp>
 #include <seqan3/range/views/take.hpp>
 #include <seqan3/range/views/take_exactly.hpp>
+
```
Comment (marehr) [inline test/unit/range/views/repeat_test.cpp:19]:
```suggestion
```

File: test/unit/range/views/single_pass_input_test.cpp [hunk 20]
```diff
@@ -17,6 +17,8 @@
 #include <seqan3/range/views/zip.hpp>
 #include <seqan3/std/ranges>
```
Comment (marehr) [inline test/unit/range/views/single_pass_input_test.cpp:19]:
```suggestion
```

File: test/unit/range/views/slice_test.cpp [hunk 21]
```diff
@@ -23,7 +23,9 @@
 #include <seqan3/std/algorithm>
 #include <seqan3/std/concepts>
 #include <seqan3/std/ranges>
+
```
Comment (marehr) [inline test/unit/range/views/slice_test.cpp:26]:
```suggestion
```

File: test/unit/range/views/take_test.cpp [hunk 22]
```diff
@@ -24,7 +24,9 @@
 #include <seqan3/std/algorithm>
 #include <seqan3/std/concepts>
 #include <seqan3/std/ranges>
+
```
Comment (marehr) [inline test/unit/range/views/take_test.cpp:27]:
```suggestion
```

File: test/unit/range/views/type_reduce_test.cpp [hunk 23]
```diff
@@ -17,7 +17,9 @@
 #include <seqan3/std/concepts>
 #include <seqan3/std/algorithm>
 #include <seqan3/std/ranges>
+
```
Comment (marehr) [inline test/unit/range/views/type_reduce_test.cpp:20]:
```suggestion
```

File: test/unit/search/configuration/hit_test.cpp [hunk 24]
```diff
@@ -10,6 +10,8 @@
 #include <seqan3/core/detail/pack_algorithm.hpp>
 #include <seqan3/search/configuration/hit.hpp>
```
Comment (marehr) [inline test/unit/search/configuration/hit_test.cpp:12]:
```suggestion
```

File: test/unit/std/concept/core_language_test.cpp [hunk 25]
```diff
@@ -12,11 +12,13 @@
 #include <seqan3/core/concept/core_language.hpp>
 #include <seqan3/std/iterator>
```
Comment (marehr) [inline test/unit/std/concept/core_language_test.cpp:14]:
```suggestion
```

File: test/unit/test/seqan2_test.cpp [hunk 26]
```diff
@@ -8,6 +8,8 @@
 #include <gtest/gtest.h>
 
 #include <seqan3/std/ranges>
+
```
Comment (marehr) [inline test/unit/test/seqan2_test.cpp:11]:
```suggestion
```

File: test/unit/utility/type_traits/function_traits_test.cpp [hunk 27]
```diff
@@ -11,6 +11,8 @@
 
 #include <seqan3/utility/type_traits/function_traits.hpp>
```
Comment (marehr) [inline test/unit/utility/type_traits/function_traits_test.cpp:13]:
```suggestion
```

**Comment by eseiler** [comment]
Did you forget to commit `seqan3/test/expect_same_type.hpp` ?

File: test/unit/alignment/matrix/alignment_coordinate_test.cpp [hunk 28]
```diff
@@ -13,27 +13,28 @@
 
 #include <seqan3/alignment/matrix/alignment_coordinate.hpp>
 #include <seqan3/test/pretty_printing.hpp>
+#include <seqan3/test/expect_same_type.hpp>
```
Comment (marehr) [inline test/unit/alignment/matrix/alignment_coordinate_test.cpp:16]:
```suggestion
#include <seqan3/test/expect_same_type.hpp>
#include <seqan3/test/pretty_printing.hpp>
```

File: test/unit/alignment/matrix/detail/affine_cell_proxy_test.cpp [hunk 29]
```diff
@@ -10,6 +10,7 @@
 #include <seqan3/alignment/matrix/detail/affine_cell_proxy.hpp>
 #include <seqan3/core/tuple/common_tuple.hpp>
 #include <seqan3/test/expect_same_type.hpp>
+#include <seqan3/test/expect_same_type.hpp>
```
Comment (marehr) [inline test/unit/alignment/matrix/detail/affine_cell_proxy_test.cpp:13]:
```suggestion
#include <seqan3/test/expect_same_type.hpp>
```

Comment (marehr) [inline test/unit/alignment/matrix/detail/affine_cell_proxy_test.cpp:13]:
```suggestion
```

File: test/unit/std/concept/core_language_test.cpp [hunk 30]
```diff
@@ -11,12 +11,13 @@
 
 #include <seqan3/core/concept/core_language.hpp>
 #include <seqan3/std/iterator>
+#include <seqan3/test/expect_same_type.hpp>
```
Comment (marehr) [inline test/unit/std/concept/core_language_test.cpp:14]:
```suggestion
```

File: test/unit/std/concept/core_language_test.cpp [hunk 31]
```diff
@@ -11,12 +11,13 @@
...
 
 #include "auxiliary.hpp"
 
 TEST(core_language_concepts, same_as)
 {
-    EXPECT_TRUE((std::same_as<int, int>));
+    EXPECT_SAME_TYPE(int, int);
```
Comment (marehr) [inline test/unit/std/concept/core_language_test.cpp:20]:
```suggestion
    EXPECT_TRUE((std::same_as<int, int>));
```

**Comment by marehr** [comment]
> Did you forget to commit `seqan3/test/expect_same_type.hpp` ?
> 
> Edit: nvm, it's already in the lib

Yep, was implemented by #2312

**Review by Irallia — APPROVED** [review]
LGFM thank you!

File: test/unit/io/alignment_file/sam_tag_dictionary_test.cpp [hunk 32]
```diff
@@ -11,6 +11,7 @@
 
 #include <seqan3/io/alignment_file/sam_tag_dictionary.hpp>
 #include <seqan3/std/concepts>
+#include <seqan3/test/expect_same_type.hpp>
```
Comment (eseiler) [inline test/unit/io/alignment_file/sam_tag_dictionary_test.cpp:14]:
```suggestion
#include <seqan3/std/concepts>

#include <seqan3/io/alignment_file/sam_tag_dictionary.hpp>
#include <seqan3/test/expect_same_type.hpp>
```

File: test/unit/range/container/small_vector_test.cpp [hunk 33]
```diff
@@ -14,6 +14,7 @@
 #include <seqan3/range/container/small_vector.hpp>
 #include <seqan3/std/ranges>
```
Comment (eseiler) [inline test/unit/range/container/small_vector_test.cpp:15]: Can you move this up, please?
Too far away to do a suggestion..

File: test/unit/range/views/drop_test.cpp [hunk 34]
```diff
@@ -24,6 +24,7 @@
 #include <seqan3/std/concepts>
 #include <seqan3/std/ranges>
```
Comment (eseiler) [inline test/unit/range/views/drop_test.cpp:25]: Also here + `std/algorithm` one line above

File: test/unit/range/views/single_pass_input_test.cpp [hunk 35]
```diff
@@ -16,6 +16,7 @@
 #include <seqan3/range/views/persist.hpp>
 #include <seqan3/range/views/zip.hpp>
 #include <seqan3/std/ranges>
```
Comment (eseiler) [inline test/unit/range/views/single_pass_input_test.cpp:18]: And here

File: test/unit/range/views/slice_test.cpp [hunk 36]
```diff
@@ -24,6 +24,7 @@
 #include <seqan3/std/concepts>
 #include <seqan3/std/ranges>
```
Comment (eseiler) [inline test/unit/range/views/slice_test.cpp:25]: = 1 line above

File: test/unit/range/views/take_test.cpp [hunk 37]
```diff
@@ -25,6 +25,7 @@
 #include <seqan3/std/concepts>
 #include <seqan3/std/ranges>
```
Comment (eseiler) [inline test/unit/range/views/take_test.cpp:26]: + line above

File: test/unit/range/views/type_reduce_test.cpp [hunk 38]
```diff
@@ -18,6 +18,7 @@
 #include <seqan3/std/algorithm>
 #include <seqan3/std/ranges>
```
Comment (eseiler) [inline test/unit/range/views/type_reduce_test.cpp:19]: + line above

File: test/unit/test/seqan2_test.cpp [hunk 39]
```diff
@@ -8,6 +8,7 @@
 #include <gtest/gtest.h>
 
 #include <seqan3/std/ranges>
+#include <seqan3/test/expect_same_type.hpp>
 #include <seqan3/test/seqan2.hpp>
```
Comment (eseiler) [inline test/unit/test/seqan2_test.cpp:12]:
```suggestion
#include <seqan3/std/ranges>

#include <seqan3/test/expect_same_type.hpp>
#include <seqan3/test/seqan2.hpp>
```

**Review by eseiler — CHANGES_REQUESTED** [review]
LGTM, but can we move the `std` includes to the correct place while touching the files?

**Review by eseiler — APPROVED** [review]

**Pull request merged**